File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamnotifications/callback/UserNotificationsCallback.java
package in.dragonbra.javasteam.steam.handlers.steamnotifications.callback;

import in.dragonbra.javasteam.protobufs.steamclient.SteammessagesClientserver2.CMsgClientUserNotifications;
import in.dragonbra.javasteam.steam.handlers.steamnotifications.Notification;
import in.dragonbra.javasteam.steam.steamclient.callbackmgr.CallbackMsg;

import java.util.ArrayList;
import java.util.List;

/**
 * Fired when the client receives user notifications.
 */
public class UserNotificationsCallback extends CallbackMsg {

    private final List<Notification> notifications;

    public UserNotificationsCallback(CMsgClientUserNotifications.Builder msg) {
        notifications = new ArrayList<>();
        for (CMsgClientUserNotifications.Notification notification : msg.getNotificationsList()) {
            notifications.add(new Notification(notification));
        }
    }

    /**
     * @return the notifications
     */
    public List<Notification> getNotifications() {
        return notifications;
    }
}

File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamnotifications/callback/UserNotificationsCallback.kt
package `in`.dragonbra.javasteam.steam.handlers.steamnotifications.callback

import `in`.dragonbra.javasteam.base.ClientMsgProtobuf
import `in`.dragonbra.javasteam.base.IPacketMsg
import `in`.dragonbra.javasteam.protobufs.steamclient.SteammessagesClientserver2.CMsgClientUserNotifications
import `in`.dragonbra.javasteam.steam.handlers.steamnotifications.Notification
import `in`.dragonbra.javasteam.steam.steamclient.callbackmgr.CallbackMsg

/**
 * Fired when the client receives user notifications.
 */
class UserNotificationsCallback(packetMsg: IPacketMsg) : CallbackMsg() {

    /**
     * Gets the notifications
     */
    val notifications: List<Notification>

    init {
        val resp = ClientMsgProtobuf<CMsgClientUserNotifications.Builder>(
            CMsgClientUserNotifications::class.java,
            packetMsg
        )

        notifications = resp.body.notificationsList.map(::Notification)
    }
}
